### ISSUE TITLE
Bounding Box format for Training and received from Inference

### ISSUE THREAD

**Issue by rroosshhaann** [keras-team/keras-io#479]
### Problem: 
When visualizing the output of the model, it appears that it is necessary to perform an xy-swap on the bounding boxes.

### Potential source:
When preprocessing the training data bounding box an xy swap is performed  
https://github.com/keras-team/keras-io/blob/5bf2b282c4e9809a028e1601611afaafd76b80dd/examples/vision/retinanet.py#L388

as well as a conversion to xy-wh box definition.
https://github.com/keras-team/keras-io/blob/5bf2b282c4e9809a028e1601611afaafd76b80dd/examples/vision/retinanet.py#L403

The issue seems to be that the conversion to xy-wh assumes x1,y1, x2, y2, as the input, 
https://github.com/keras-team/keras-io/blob/5bf2b282c4e9809a028e1601611afaafd76b80dd/examples/vision/retinanet.py#L90-L91

however this processing is done after the xy-swap. 

### Symptom:
On inference the visualization assumes the output is of the type x1, y1, x2, y2.  However I noticed that my boxes seemed presented flipped along the xy axis. 

When I modified the visualization to assume y1, x1, y2, x2, the visualization was improved. 

### Potential Fix?
I have removed the xy-swap in my preprocessing and now learning has GREATLY improved.

**Comment by srihari-humbarwadi** [comment]
The `swap_xy` function as the name suggests, swaps the x, y ordering for a bounding box.
There is no change required to the example since `tensorflow_datasets` stores bounding boxes as `ymin, xmin, ymax, xmax` as described here https://www.tensorflow.org/datasets/api_docs/python/tfds/features/BBox

**Comment by rroosshhaann** [comment]
Thank you.  I understand what the function does and why. I may not be clear explaining the problem. 

The flow of code in the data prep process, first does an xy swap, but then undertakes an xy_wh change, which as an input expects x and y have not yet been swapped.

**Comment by srihari-humbarwadi** [comment]
The example uses two formats for representing boxes ---
 - `x, y, w, h`
 - `xmin, ymin, xmax, ymax`

However as mentioned above, `tensorflow_datasets` follows `ymin, xmin, ymax, xmax` format.  For this reason, we swap the ordering of the coordinates when parsing data coming from `tensorflow_datasets` loader.

 If you are not using `tensorflow_datasets`you can either remove the call to `swap_xy` or store your data in the format followed by `tensorflow_datasets`.

**Comment by sachinprasadhs** [comment]
Hi,

If your concern is addressed by the above comment, feel free to close the issue.

For latest update on the `retinanet` example, please follow the tutorial here https://keras.io/examples/vision/retinanet/#implementing-anchor-generator
